docker/docker.hcl

```diff
@@ -0,0 +1,23 @@
+group "default" {
+  targets = ["graphql-mesh"]
+}
+
+variable "PWD" {
+  default = "."
+}
+
+variable "context" {
+  default = "${PWD}/docker"
+}
+
+target "graphql-mesh" {
+  context = "${context}"
+  dockerfile = "${PWD}/docker/Dockerfile"
+  platforms = ["linux/amd64", "linux/arm64"]
+  tags = [
+    "ghcr.io/ardatan/graphql-mesh:v1"
```

Comment:
I don't think this should be configured with this tag here. 

At the moment, we should push this with the following conditionals:
- PRs should use the commit ID as image tag. 
- `main` should also use `commit` ID and also `:v1-preview`